### PR TITLE
DumpStore with list of ids.

### PR DESCRIPTION
**Pull request by martinfurmanski**
Syntax like: DumpStore file:id1,id2

Allows dumping just a selected set of ids.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java [hunk 1]
```diff
@@ -175,43 +192,72 @@ public final void dump( STORE store ) throws Exception
...
+            }
+        }
+        else
         {
-            RECORD record = store.forceGetRecord( i );
+            out.print( record );
+            buffer.clear();
+            fileChannel.read( buffer, id * size );
+            buffer.flip();
             if ( record.inUse() )
```
Comment (digitalstain) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java:253]: If i read this correctly, this 3-clause if can be reduced to 2, if we consolidate the first if with the catch all else, correct?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java [hunk 2]
```diff
@@ -175,43 +192,72 @@ public final void dump( STORE store ) throws Exception
...
             {
-                used++;
+                dumpHex( buffer, id * size );
             }
-            Object transform = transform( record );
-            if ( transform != null )
+            else if ( allZero( buffer ) )
```
Comment (digitalstain) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java:257]: How easy would it be to aggregate all zero records in one statement? Something like 
- while it's all zero
  - keep it in a buffer
- if we encounter a non zero record or the zero buffer becomes full
  - dump and empty the buffer, if non empty
  - print the record

**Comment by digitalstain** [comment]
apart from the comments, which could be implemented at a later time, this looks ok

File: community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java (hunk 2 above)
Comment (martinfurmanski) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java:257]: I like this idea, but hopefully this path is rarely taken. This is the "raw" format when there is no suitable transform function.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java (hunk 1 above)
Comment (martinfurmanski) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java:253]: Yes that could be done, but then the allZero( buffer ) check must be first. I think inUse() is more expected and it is cheaper.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java [hunk 3]
```diff
@@ -43,29 +43,53 @@ public static void main( String... args ) throws Exception
         for ( String arg : args )
         {
             File file = new File( arg );
-            if ( !file.isFile() )
+            long[] ids = null; // null means all possible ids
+
+            if( file.isFile() )
+            {
+                /* If file exists, even with : in its path, then accept it straight off. */
```
Comment (martinfurmanski) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java:50]: I updated the logic to handle cases with : in the path better.

**Pull request merged**